kimhaejung/app.js

```diff
@@ -35,6 +36,89 @@ app.get("/ping", function (req, res, next) {
   res.json({ message: "pong" });
 });
 
+app.post("/users/signup", async (req, res, next) => {
+  const { name, email, profileImage, password } = req.body;
+
+  await appDataSource.query(
+    `INSERT INTO users(
+      name,
+      email,
+      profile_image,
+      password
+    ) VALUES (?, ?, ?, ?);
+    `,
+    [name, email, profileImage, password]
+  );
+  res.status(201).json({ message: "userCreated" });
+});
+
+app.post("/posts", async (req, res, next) => {
+  const { title, content, userId, imageUrl } = req.body;
+
+  await appDataSource.query(
+    `INSERT INTO posts(
+      title,
+      content,
+      user_id,
+      image_url
+    ) VALUES (?, ?, ?, ?);
+    `,
+    [title, content, userId, imageUrl]
+  );
+  res.status(201).json({ message: "postCreated" });
+});
+
+app.get("/posts/look", async (req, res, next) => {
+  await appDataSource.query(
+    `SELECT
+      users.id as userId,
+      users.profile_image as userProfileImage,
+      posts.id as postingId,
+      posts.image_url as postingImageUrl,
+      posts.content as postingContent
+      FROM users
+      INNER JOIN posts ON users.id = posts.user_id`,
+    (err, rows) => {
+      res.status(200).json({ data: rows });
+    }
+  );
```

Comment:
유저 정보가 있는 게시글만 조회할 수 있도록 JOIN을 잘 사용해주신 것 같아요!

- 지금 `query()` 함수의 두번째 인자로 callback 함수를 실행하여 response 객체를 반환하고 있는데 아래와 같이 작성해보고 console에서 어떤 값이 출력되는지도 확인해보면 좋을 것 같습니다.
callback 함수를 계속 중첩하다 보면, 가독성이 떨어지고 관리하기도 어려워지기 때문입니다.
async-await 키워드를 활용하여 callback 함수의 중첩을 줄일 수도 있습니다!
- 다소 마이너한 요소일 수는 있지만, sql문의 포맷을 일정하게 맞추어 작성하는 방법도 가독성을 높이고 유지, 보수를 용이하게 하는 요인이 될 수 있습니다. 아래 예시를 참고하셔서 sql 포맷도 수정해보면 좋을 것 같아요!
다만, 해당 방법이 반드시 정답은 아니고, 위코드에서는 아래와 같은 컨벤션을 따르고 있습니다.
회사마다 다르다는 점도 참고해주세요~!
```suggestion
// SELECT, FROM과 같은 키워드는 대문자로 표기해 주세요.
  const result = await appDataSource.query(
    `SELECT
      users.id as userId,
      users.profile_image as userProfileImage,
      posts.id as postingId,
      posts.image_url as postingImageUrl,
      posts.content as postingContent
    FROM users
    INNER JOIN posts ON users.id = posts.user_id`,
    }
  );
  console.log(result)
  return res.status(200).json({ data: result });
```